contracts/GatewayComposer.sol

```diff
@@ -376,4 +382,36 @@ contract GatewayComposer {
         success_ = true;
     }
 
+    /**
+     * @notice Remove storage & code from blockchain.
+     *
+     * @dev Function requires:
```

Comment:
Since revert can only be called via GC, should there also be a check that there is no stake in Gateway?

Reply:
correct, but it is not possible to exhaustively check (unless if we would store the stake message hashes)